middleware/timeout/timeout.go

```diff
@@ -21,42 +22,34 @@ func New(h fiber.Handler, config ...Config) fiber.Handler {
 
 		timeout := cfg.Timeout
 		if timeout <= 0 {
-			return runHandler(ctx, h, cfg)
+			return h(ctx)
 		}
 
-		parent := ctx.Context()
-		tCtx, cancel := context.WithTimeout(parent, timeout)
-		ctx.SetContext(tCtx)
-		defer func() {
-			cancel()
-			ctx.SetContext(parent)
+		err := make(chan error, 1)
+		go func() {
```

Comment:
Why?

Reply:
If the timeout occurs, the go routine will otherwise block forever because there is nothing to receive from the channel. Hence the buffered channel.